### PR TITLE
Update tech selections

### PR DESCRIPTION
**Pull request by JoshuaHinman**
# Description

- This PR revises the PATCH /teams/techs/:techId route to only update a single tech stack selection, setting the `isSelected` value
- Also, a permission bug has been fixed for POST /teams/{teamId}/techs , ensuring users can only add a new tech for thier own team

## Issue link

Fixes # [86b1ym631](https://app.clickup.com/t/86b1ym631)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [x] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- migrate db for updates
- yarn test

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: src/techs/techs.controller.ts [hunk 1]
```diff
@@ -477,24 +476,17 @@ export class TechsController {
...
-    })
     @CheckAbilities({ action: Action.Update, subject: "TeamTechStackItem" })
-    @Patch("teams/:teamId/techs/selections")
+    @Patch("teams/techs/:techId")
     updateTechStackSelections(
         @Request() req,
-        @Param("teamId", ParseIntPipe) teamId: number,
-        @Body(ValidationPipe) updateTechSelectionsDto: UpdateTechSelectionsDto,
+        @Param("techId", ParseIntPipe) techId: number,
+        @Body(ValidationPipe) techSelectionDto: UpdateTechSelectionDto,
```
Comment (cherylli) [inline src/techs/techs.controller.ts:484]: should we leave it as updateTechSelectionDto just to be more clear that we are using the UpdateTechSelectionDto? And also consistent with the function `techs.service.ts`

File: src/techs/techs.service.ts [hunk 2]
```diff
@@ -74,46 +74,53 @@ export class TechsService {
...
+                id: techId,
+            },
+            select: {
+                categoryId: true,
+                voyageTeamId: true,
+            },
+        });
+        if (!tech) {
+            throw new NotFoundException(
+                `Tech ${MAX_SELECTION_COUNT} not found.`,
```
Comment (cherylli) [inline src/techs/techs.service.ts:92]: is this supposed to be techId instead of `MAX_SELECTION_COUNT`?

File: src/techs/techs.service.ts [hunk 3]
```diff
@@ -124,8 +131,14 @@ export class TechsService {
         //check for valid teamId
         await this.validateTeamId(teamId);
 
+        //check if user is a member of the team
         manageOwnVoyageTeamWithIdParam(req.user, teamId);
```
Comment (cherylli) [inline src/techs/techs.service.ts:135]: I think we won't need `manageOwnVoyageTeamWithIdParam` if we have `userCanChangeCategory`
Also could we move `userCanChangeCategory` to the `ability` folder?

File: src/techs/techs.service.ts [hunk 4]
```diff
@@ -353,7 +366,8 @@ export class TechsService {
         techStackCategoryId: number,
         updateTechStackCategoryDto: UpdateTechStackCategoryDto,
     ) {
-        const permission = await this.userCanChangeCategory(
+        const permission = true;
```
Comment (cherylli) [inline src/techs/techs.service.ts:369]: how come permission is set to true but then a few lines below there is code to throw an error when permission is not truthy?

File: test/techs.e2e-spec.ts [hunk 5]
```diff
@@ -677,142 +677,113 @@ describe("Techs Controller (e2e)", () => {
         });
     });
 
-    describe("PATCH voyages/teams/:teamId/techs/selections - updates isSelected value of tech stack items", () => {
-        it("should return 200 and an array of updated techs, if successful", async () => {
-            const teamId: number = 2;
+    describe("PATCH voyages/teams/:teamId/techs/selections - updates isSelected value of a tech stack tems", () => {
```
Comment (cherylli) [inline test/techs.e2e-spec.ts:680]:
```suggestion
    describe("PATCH voyages/teams/:teamId/techs/selections - updates isSelected value of a tech stack item", () => {
```

File: src/techs/techs.service.ts (hunk 4 above)
Comment (JoshuaHinman) [reply src/techs/techs.service.ts:369]: oops - I did this as a shortcut when refactoring and forgot about it...

File: src/techs/techs.service.ts (hunk 3 above)
Comment (JoshuaHinman) [reply src/techs/techs.service.ts:135]: I tried removing this, but got a testing error.  Looking at it closer, these functions are comparing different values.  `manageOwnVoyageTeamWithIdParam` verifies that user is in the team from the URI params, and `userCanChangeCategory` verifies that the user is in the team from `category.voyageTeamId`

Comment (cherylli) [reply src/techs/techs.service.ts:135]: but I mean if the user is in the team in `category.voyageTeamId` then we can assume they are allowed to update it, without having to check if the user is in the team from the uri params? I guess we only need to check the one which is more restrictive

Comment (JoshuaHinman) [reply src/techs/techs.service.ts:135]: Ok - took it out and fixed the test

**Review by curtwl — APPROVED** [review]
lgtm

**Review by cherylli — APPROVED** [review]
Did some testing on swagger and ran all the tests, working as expected

**Pull request merged**